docs2/site/docs/analyzers/GQL006_InvalidInputField.md

```diff
@@ -0,0 +1,115 @@
+# GQL006: Invalid input field
+
+|                        | Value   |
+| ---------------------- | ------- |
+| **Rule ID**            | GQL006  |
+| **Category**           | Usage   |
+| **Default severity**   | Warning |
+| **Enabled by default** | Yes     |
+| **Code fix provided**  | No      |
+
+## Cause
+
+This rule triggers when a field defined on a type deriving from `InputObjectGraphType<TSourceType>` cannot be mapped to the `TSourceType` type.
+
+## Rule description
+
+This diagnostic is reported when the input field name cannot be mapped to any field, property, or constructor parameter of the `TSourceType` type. The name comparison is case-insensitive. The input field can be mapped when all the following rules are respected:
+
+### Mapping rules for property/field
+
+- Must have the exact same name (case-insensitive)
+- Must be public
+- Must not be static
+- Must be settable (have a public setter / not a constant)
```

Comment:
Currently, I believe the `ToObject` code allows deserializing to get-only properties that are set via a constructor argument.  Have you checked that?

Reply:
Oh, I see you have 'rules for constructor parameter' below, so I assume you have.

Reply:
Btw, it should be possible to replace the `ToObject` implementation with a source generator similar to how MS did with a [Configure](https://andrewlock.net/exploring-the-dotnet-8-preview-using-the-new-configuration-binder-source-generator/#looking-at-the-generated-code) method in .NET 8. It will remove the use of reflection and significantly improve the deserialization [performance](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#configuration).